Realm/Realm/Schema/ObjectSchema.cs

```diff
@@ -59,12 +69,22 @@ public class ObjectSchema : IReadOnlyCollection<Property>
         /// Gets a value indicating whether this <see cref="ObjectSchema"/> describes an embedded object.
         /// </summary>
         /// <value><c>true</c> if the schema pertains to an <see cref="EmbeddedObject"/> instance; <c>false</c> otherwise.</value>
+        [Obsolete("Check against RealmSchemaType instead.")]
         public bool IsEmbedded { get; }
 
-        private ObjectSchema(string name, bool isEmbedded, IDictionary<string, Property> properties)
+        // TODO andrea: review this documentation
+        /// <summary>
+        /// Gets a <see cref="ObjectSchemaType"/> indicating whether this <see cref="ObjectSchema"/> describes
+        /// a top level object, an embedded object or an asymmetric object.
+        /// </summary>
+        /// <value>The type of ObjectSchema.</value>
+        public ObjectSchemaType RealmSchemaType { get; }
```

Comment:
The name of this property is kind of weird. I'd suggest just `Type` or `ObjectType`.